src/api/app/validators/scm_webhook_event_validator.rb

```diff
@@ -49,8 +49,8 @@ def valid_gitlab_event?
   end
 
   def valid_push_event?
-    return true if @record.payload.fetch(:ref, '').start_with?('refs/heads/')
+    return true if @record.payload.fetch(:ref, '').start_with?('refs/heads/', 'refs/tags/')
 
-    @record.errors.add(:base, 'Push event supported only for branches.')
+    @record.errors.add(:base, 'Push event supported only for branches and tags.')
```

Comment:
:thought_balloon: Is there any other case for pushing, apart from branches or tags?

Reply:
References are only for branches or tags, but we never know if someone tinkers with the payload. This is to ensure that we're dealing with a proper payload for either a branch or a tag.